inflate.c

```diff
@@ -359,6 +378,33 @@ void makefixed(void) {
 }
 #endif /* MAKEFIXED */
 
+int ZLIB_INTERNAL inflate_ensure_window(struct inflate_state *state)
```

Comment:
Splitting this out to a separate function should be a separate preparation commit (same PR). Makes it easier to bisect and benchmark.